cpp/do_module/do_module.cpp

```diff
@@ -0,0 +1,391 @@
+    } else if (map_item.value.IsBool()) {
+      new_params.Insert(map_item.key, mg::Value(map_item.value.ValueBool()));
+    } else if (map_item.value.IsString()) {
+      new_params.Insert(map_item.key, mg::Value(map_item.value.ValueString()));
+    } else if (map_item.value.IsInt()) {
+      new_params.Insert(map_item.key, mg::Value(map_item.value.ValueInt()));
+    } else if (map_item.value.IsDouble()) {
+      new_params.Insert(map_item.key, mg::Value(map_item.value.ValueDouble()));
+    } else {
+      // Temporal types and paths not yet supported
+      throw std::runtime_error("Can't parse some of the arguments!");
+    }
+  }
+
+  return new_params;
+}
+
+mg::Client::Params GetClientParams() {
+  auto *host = kDefaultHost;
+  auto port = kDefaultPort;
+  auto *username = "";
+  auto *password = "";
+
+  auto *maybe_host = std::getenv(kMgHost);
+  if (maybe_host) {
+    host = std::move(maybe_host);
+  }
+
+  const auto *maybe_port = std::getenv(kMgPort);
+  if (maybe_port) {
+    port = static_cast<uint16_t>(std::move(*maybe_port));
+  }
+
+  const auto *maybe_username = std::getenv(kMgUsername);
+  if (maybe_username) {
+    username = std::move(maybe_username);
+  }
+
+  const auto *maybe_password = std::getenv(kMgPassword);
+  if (maybe_password) {
+    password = std::move(maybe_password);
+  }
+
+  return mg::Client::Params{.host = std::move(host),
+                            .port = std::move(port),
+                            .username = std::move(username),
+                            .password = std::move(password)};
+}
+
+QueryResults ExecuteQuery(const std::string &query, const mgp::Map &query_parameters) {
+  mg::Client::Init();
+
+  auto client = mg::Client::Connect(GetClientParams());
+
+  if (!client) {
+    throw std::runtime_error("Unable to connect to client!");
+  }
+
+  auto param_names = ExtractParamNames(query_parameters);
+  auto preffix_query = ConstructQueryPreffix(param_names);
+  auto final_query = ConstructFinalQuery(query, preffix_query);
+
+  const auto final_parameters = ConstructParams(param_names, query_parameters);
+
+  if (!client->Execute(final_query, final_parameters.AsConstMap())) {
+    throw std::runtime_error("Error while executing do module!");
+  }
+
+  auto columns = client->GetColumns();
+
+  std::vector<std::vector<mg::Value>> results;
+  while (const auto maybe_result = client->FetchOne()) {
+    if ((*maybe_result).size() == 0) {
+      break;
+    }
+
+    auto result = *maybe_result;
+    results.push_back(std::move(result));
+  }
+
+  client->DiscardAll();
+
+  mg::Client::Finalize();
+
+  return QueryResults{.columns = std::move(columns), .results = std::move(results)};
+}
+
+void InsertConditionalResults(const mgp::RecordFactory &record_factory, const QueryResults &query_results) {
+  for (const auto &row : query_results.results) {
+    auto result_map = mgp::Map();
```

Comment:
prefer initialization to assignment:

```
mgp::Map result_map;
```

Reply:
FYI that is construction, no assignment is involved. see https://godbolt.org/z/8P78MeWrW

Reply:
@Ignition I thought all the time it was an assignment, thanks :+1: